wire-runtime/src/main/java/com/squareup/wire/MessageAdapter2.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2015 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import okio.Buffer;
+import okio.BufferedSink;
+import okio.BufferedSource;
+import okio.Okio;
+
+/** An adapter for reading and writing Wire messages. */
+// NOTE: This class is suffixed with a "2" for git's sake. It is renamed in the next commit.
+public abstract class MessageAdapter2<M extends Message> {
```

Comment:
Probably. This was an interface before the convenience methods were added. Also I wanted to start small but it kind of bubbled into a bigger change.